starter_code/app.js

```diff
@@ -1,26 +1,100 @@
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
+const SpotifyWebApi = require("spotify-web-api-node");
 
 // require spotify-web-api-node package here:
 
+const clientId = "285343778767412c89c8a3549afd7842",
+  clientSecret = "f72c74c9e24e457dac6393c6da78c45a";
 
+const spotifyApi = new SpotifyWebApi({
+  clientId: clientId,
+  clientSecret: clientSecret
+});
 
-const app = express();
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then(data => {
+    spotifyApi.setAccessToken(data.body["access_token"]);
+  })
+  .catch(error => {
+    console.log("Something went wrong when retrieving an access token", error);
+  });
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+const app = express();
+app.use(express.static(__dirname + "/public"));
+// app.use(express.static("public"));
+app.set("view engine", "hbs"); // don't get it !!!!!!
+app.set("views", __dirname + "/views");
+
+app.get("/", (req, res) => {
+  // console.log(req.query)
+  // console.log(req.body)
+  // const data = {
+  //   title: "Contact"
+  // };
+  res.render("index");
+});
+
+app.get("/artists", (req, res) => {
+  console.log(req.query);
+
+  spotifyApi
+    .searchArtists(req.query.artist)
+    .then(data => {
+      console.log("The received data from the API: ", data.body);
+      // arr.forEach(item => {
+      console.log(data.body.artists.items[0].images[0].url);
+      //   item.images.forEach(image => {
+      //     console.log(image.url);
+      //   })
+      // });
+      // res.json(data.body)
+      res.render("artists", { artists: data.body.artists.items });
+      // ----> 'HERE WHAT WE WANT TO DO AFTER RECEIVING THE DATA FROM THE API'
+    })
+    .catch(err => {
+      console.log("The error while searching artists occurred: ", err);
+    });
+});
+
+app.get("/albums/:artistId", (req, res, next) => {
+  console.log(req.params);
```

Comment:
another syntax here would have been to deconstruct and only pass artistId as an argument:
`const { artistId } = req.params
.getArtistAlbums(artistId)
`
it is the same, just another way of doing it !